### ISSUE TITLE
Delphi 10.2.1 32-bit build attached

### ISSUE THREAD

**Issue by dslutej** [Lecrapouille/Ecstasy#4]
Here you go, Ecstasy built with Delphi 10.2.1, small quick fixes to get it to build. Tested on Windows 10 64-bit. 

[Ecstasy.zip](https://github.com/Lecrapouille/Ecstasy/files/1385697/Ecstasy.zip)

**Comment by Lecrapouille** [comment]
Hi @dslutej I do not have window 10 :) so thanks for your support. That's very kind. I'll use your zip tomorrow

**Comment by Lecrapouille** [comment]
@dslutej I looked at your zip file, there is only the exe file. I thought you included your quick fixes on it. You can submit them either on PR or give me a zip. Anyway I try to execute your exe on my Windows XP (32bits) but it failed when linking the dsound.dll during the middle of the loading process. I just noticed that the loading text is truncated.

**Comment by Lecrapouille** [comment]
PS: oh ok now it's working, I just unchecked "enabling sound" :)

**Comment by dslutej** [comment]
Oops, my bad. Here are the missing pas files as well. Yeah, sounds I have ignored. :-D

[src.zip](https://github.com/Lecrapouille/Ecstasy/files/1388822/src.zip)

**Comment by Lecrapouille** [comment]
Hi @dslutej ! I cannot apply all your fixes (specialy concerning DirectX) it does not compile on Delphi5. What's happening when compiling on Delphi 10 with the original code ? Is it compile with warnings or just failing ? I'm thinking to add some conditional code to check on which Delphi version you have. But I not sure how to do that (make long time I did not program in Delphi).

**Comment by dung059** [comment]
> Here you go, Ecstasy built with Delphi 10.2.1, small quick fixes to get it to build. Tested on Windows 10 64-bit.
> 
> [Ecstasy.zip](https://github.com/Lecrapouille/Ecstasy/files/1385697/Ecstasy.zip)


I change some code and complie is OK, not check error!!!
[src.zip](https://github.com/Lecrapouille/Ecstasy/files/2974841/src.zip)

**Comment by Lecrapouille** [comment]
@dung059 Thank you for your commit :) Unfortunately I'm still stuck on Windows XP and Delphi 5 (I have to upgrade everything) and I have compilation errors with your patch (my Delphi knowledges get rusty) so I'll rename my master branch or push your change in special branch. What configuration do you have ?

**Comment by Lecrapouille** [comment]
@dung059 you are right I noticed a regression loading no cars. I dunno this was working before. Quick and dirty patched on master
